src/backends/torch/torchlib.cc

```diff
@@ -1110,83 +1187,129 @@ namespace dd
+                    results_ad.add("nclasses", (int)_nclasses);
 
-                results_ads.push_back(results_ad);
+                    results_ads.push_back(results_ad);
+                  }
               }
-          }
-        else if (_timeserie)
-          {
-            output = output.to(cpu);
-            auto output_acc = output.accessor<float, 3>();
-            for (int j = 0; j < output.size(0); ++j)
+            else if (_timeserie)
               {
-                std::vector<APIData> series;
-                for (int t = 0; t < output.size(1); ++t)
+                output = output.to(cpu);
+                auto output_acc = output.accessor<float, 3>();
+                for (int j = 0; j < output.size(0); ++j)
                   {
-                    std::vector<double> preds;
-                    for (unsigned int k = 0; k < this->_inputc._ntargets; ++k)
+                    std::vector<APIData> series;
+                    for (int t = 0; t < output.size(1); ++t)
                       {
-                        preds.push_back(output_acc[j][t][k]);
+                        std::vector<double> preds;
+                        for (unsigned int k = 0; k < this->_inputc._ntargets;
+                             ++k)
+                          {
+                            preds.push_back(output_acc[j][t][k]);
+                          }
+                        APIData ts;
+                        ts.add("out", preds);
+                        series.push_back(ts);
                       }
-                    APIData ts;
-                    ts.add("out", preds);
-                    series.push_back(ts);
+                    APIData result_ad;
+                    if (!inputc._ids.empty())
+                      result_ad.add("uri", inputc._ids.at(nsample++));
+                    else
+                      result_ad.add("uri", std::to_string(nsample++));
+                    result_ad.add("series", series);
+                    result_ad.add("probs",
+                                  std::vector<double>(series.size(), 1.0));
+                    result_ad.add("loss", 0.0);
+                    results_ads.push_back(result_ad);
                   }
-                APIData result_ad;
-                if (!inputc._ids.empty())
-                  result_ad.add("uri", inputc._ids.at(nsample++));
-                else
-                  result_ad.add("uri", std::to_string(nsample++));
-                result_ad.add("series", series);
-                result_ad.add("probs",
-                              std::vector<double>(series.size(), 1.0));
-                result_ad.add("loss", 0.0);
-                results_ads.push_back(result_ad);
               }
           }
       }
-    outputc.add_results(results_ads);
 
-    if (_timeserie)
-      out.add("timeseries", true);
-    outputc.finalize(output_params, out,
-                     static_cast<MLModel *>(&this->_mlmodel));
+    if (extract_layer.empty())
```

Comment:
Suggestion for more clarity:
```cpp
bool supervised_output = extract_layer.empty();

if (supervised_output) { ...
```

Reply:
We talked about refactoring supervised / unsupervised output at some point, so maybe the notion of supervised_output will disappear in the near future, I am not sure to want to put it here